File: cs50/cs50/unit6/sentimental/caesar/caesar.py
import sys

# Check args
if len(sys.argv) != 2:
    print("Usage: ./caesar [key]")
    sys.exit(1)
else:
    # Set k and p values
    k = int(sys.argv[1])
    p = input("plaintext: ")

    # ci=(pi+k)mod26
    ctext = ""
    for ch in p:
        if ch.isupper():
            ctext += chr(((ord(ch) + k) - ord('A')) % 26 + ord('A'))
        elif ch.islower():
            ctext += chr(((ord(ch) + k) - ord('a')) % 26 + ord('a'))
        else:
            ctext += ch

    print("ciphertext: "+ctext)

File: random-projects/c/cs50/unit2/caesar/caesar.c
#include <stdio.h>
#include <cs50.h>
#include <ctype.h>
#include <stdlib.h>
#include <string.h>

int main(int argc, string argv[])
{
    if (argc != 2)
    {
        printf("Usage: ./caesar [key]");
        return 1;
    }
    else
    {
        int k = atoi(argv[1]);
        string p = get_string("plaintext: ");

        printf("ciphertext: ");
        //ci=(pi+k)mod26
        for (int i = 0; i < strlen(p); i++)
        {
            if (isupper(p[i]))
            {
                printf("%c", ((p[i] + k) - 'A') % 26 + 'A');
            }
            else if (islower(p[i]))
            {
                printf("%c", ((p[i] + k) - 'a') % 26 + 'a');
            }
            else
            {
                printf("%c", p[i]);
            }
        }

        printf("\n");
    }
}